.github/workflows/build.yml

```diff
@@ -372,17 +175,32 @@ jobs:
         run: cargo add --dev --features bindgen aws-lc-sys --package rustls --verbose && cargo install bindgen-cli --verbose
       - run: cross test --package rustls --target ${{ matrix.target }}
 
-  semver:
-    name: Check semver compatibility
+  cross-with-rc-alias:
+    # XXX TBD FIND A WAY TO TEST WITH THIS KIND OF EMBEDDED TARGET
+    name: cross-target build testing with RC alias
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+          # 32-bit target with no atomic ptr
+          - thumbv6m-none-eabi
+          # XXX TBD other targets to test this build on ???
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4
         with:
           persist-credentials: false
 
-      - name: Check semver
-        uses: obi1kenobi/cargo-semver-checks-action@v2
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+      - name: Install cross (cross-rs) from GitHub
+        run: cargo install cross --git https://github.com/cross-rs/cross
+      # XXX TODO FIND WAY TO GET BUILD WORKING WITH BUILT-IN PROVIDERS
+      - run: cross build --no-default-features --features custom-provider,withrcalias --package rustls --target ${{ matrix.target }}
+      # XXX TRY with alloc::sync::Arc - THIS SHOULD FAIL DUE TO LACK OF SUPPORT for thumbv6m:
+      - run: cross build --no-default-features --features custom-provider --package rustls --target ${{ matrix.target }}
```

Comment:
FAILED AS EXPECTED due to missing support for `alloc::sync` on `thumbv6m` - https://github.com/rustls/rustls/actions/runs/10659143604/job/29541249343?pr=2088#step:6:15 - time to remove these lines:

```suggestion
```